src/Components/Components/src/RouteAttribute.cs

```diff
@@ -8,7 +8,7 @@ namespace Microsoft.AspNetCore.Components
     /// <summary>
     /// Indicates that the associated component should match the specified route template pattern.
     /// </summary>
-    [AttributeUsage(AttributeTargets.Class, AllowMultiple = true, Inherited = true)]
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = true, Inherited = false)]
     public class RouteAttribute : Attribute
```

Comment:
Make it sealed? I think you're allowed to redefine `AttributeUsage` on derived types so you could get in to the same problem.

Reply:
Hmm that's wierd that we didn't do this already. Since github didn't show your comment until after I merged, I'm going to add this to the API review item.